sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ChunkedDownloadUtils.java

```diff
@@ -0,0 +1,116 @@
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.storage.blob.models.BlobDownloadAsyncResponse;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.ParallelTransferOptions;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.util.function.Tuple3;
+
+import java.util.function.Function;
+
+import static java.lang.StrictMath.toIntExact;
+
+/**
+ * This class provides helper methods for lazy/chunked download.
+ *
+ * RESERVED FOR INTERNAL USE.
+ */
+public class ChunkedDownloadUtils {
+
+    /*
+    Download the first chunk. Construct a Mono which will emit the total count for calculating the number of chunks,
+    access conditions containing the etag to lock on, and the response from downloading the first chunk.
+     */
+    public static Mono<Tuple3<Long, BlobRequestConditions, BlobDownloadAsyncResponse>> getSetupMono(BlobRange range,
```

Comment:
Should this be called `downloadFirstChunk`?

Reply:
Yeah I can rename - I like that name.